grayscale-simd/videoLuma.cpp

```diff
@@ -158,50 +158,85 @@ static void arVideoLumaRGBAtoL_Emscripten_simd128(uint8_t *__restrict dest,
     v128_t pixels4_7_h =
         wasm_i16x8_shuffle(pixels4_7, pixels4_7, 1, 3, 5, 7, 9, 11, 13, 15);
     */
+    /*__m128i pixels0_3_l = _mm_unpacklo_epi8(
+        pixels0_3,
+        _mm_setzero_si128()); // pixels0_3_l =
+                              // 00[A1]00[B1]00[G1]00[R1]00[A0]00[B0]00[G0]00[R0].
+    __m128i pixels0_3_h = _mm_unpackhi_epi8(
+        pixels0_3,
+        _mm_setzero_si128()); // pixels0_3_h =
+                              // 00[A3]00[B3]00[G3]00[R3]00[A2]00[B2]00[G2]00[R2].
+    __m128i pixels4_7_l = _mm_unpacklo_epi8(
+        pixels4_7,
+        _mm_setzero_si128()); // pixels4_7_l =
+                              // 00[A5]00[B5]00[G5]00[R5]00[A4]00[B4]00[G4]00[R4].
+    __m128i pixels4_7_h = _mm_unpackhi_epi8(
+        pixels4_7,
+        _mm_setzero_si128()); // pixels4_7_h =
+                              // 00[A7]00[B7]00[G7]00[R7]00[A6]00[B6]00[G6]00[R6].*/
     v128_t pixels0_3_l = unpacklo_epi8(pixels0_3, wasm_i64x2_const(0,0));
     v128_t pixels0_3_h = unpackhi_epi8(pixels0_3, wasm_i64x2_const(0,0));
     v128_t pixels4_7_l = unpacklo_epi8(pixels4_7, wasm_i64x2_const(0,0));
     v128_t pixels4_7_h = unpackhi_epi8(pixels4_7, wasm_i64x2_const(0,0));
 
     // Multiply and add the RGB components
+    /*__m128i y0_3_l = _mm_madd_epi16(pixels0_3_l, RGBScale);
+    __m128i y0_3_h = _mm_madd_epi16(pixels0_3_h, RGBScale);
+    __m128i y4_7_l = _mm_madd_epi16(pixels4_7_l, RGBScale);
+    __m128i y4_7_h = _mm_madd_epi16(pixels4_7_h, RGBScale);*/
     v128_t y0_3_l = wasm_i32x4_dot_i16x8(pixels0_3_l, RGBScale);
     v128_t y0_3_h = wasm_i32x4_dot_i16x8(pixels0_3_h, RGBScale);
     v128_t y4_7_l = wasm_i32x4_dot_i16x8(pixels4_7_l, RGBScale);
     v128_t y4_7_h = wasm_i32x4_dot_i16x8(pixels4_7_h, RGBScale);
 
     // Horizontal add the result
+    /*__m128i y0_3 = _mm_hadd_epi32(y0_3_l, y0_3_h);
+    __m128i y4_7 = _mm_hadd_epi32(y4_7_l, y4_7_h);*/
     v128_t y0_3 = hadd_epi32(y0_3_l, y0_3_h);
```

Comment:
Probably the source of the issue is how i implemented hadd_epi32. Changing the two shuffles bits order determine differents results.

Reply:
with commit https://github.com/kalwalt/Emscripten-sketches/pull/2/commits/858d3903dc8e3c0804dc4178bb906cbb66a70e2a i solved in this way  
```c++
v128_t hadd_epi32(v128_t a, v128_t b) {
  v128_t shuffled_a = wasm_i32x4_shuffle(a, b, 0, 2, 4, 6);
  v128_t shuffled_b = wasm_i32x4_shuffle(a, b, 1, 3, 5, 7);
  return wasm_i32x4_add(shuffled_a, shuffled_b);
}
```
though i think it's not 100% correct i will merge the code. I think `__mm_hadd_epi32` should be avoided.